packages/plugin-hardhat/src/deploy-beacon.ts

```diff
@@ -0,0 +1,33 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { ContractFactory, Contract } from 'ethers';
+
+import { Manifest, BeaconDeployment } from '@openzeppelin/upgrades-core';
+
+import { Options, deploy, DeployTransaction, getUpgradeableBeaconFactory, deployBeaconImpl } from './utils';
+import { FormatTypes } from 'ethers/lib/utils';
+
+export interface DeployBeaconFunction {
+  (ImplFactory: ContractFactory, opts?: Options): Promise<Contract>;
+}
+
+export function makeDeployBeacon(hre: HardhatRuntimeEnvironment): DeployBeaconFunction {
+  return async function deployBeacon(ImplFactory: ContractFactory, opts: Options = {}) {
+    const { provider } = hre.network;
+    const manifest = await Manifest.forNetwork(provider);
+
+    const { impl } = await deployBeaconImpl(hre, ImplFactory, opts);
+
+    const UpgradeableBeaconFactory = await getUpgradeableBeaconFactory(hre, ImplFactory.signer);
+    const abi = ImplFactory.interface.format(FormatTypes.json);
```

Comment:
Agreed. This can be changed to `FormatTypes.minimal` to get a [minimal array](https://docs.ethers.io/v5/api/utils/abi/interface/#Interface--formatting).

Reply:
Addressed in https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/456/commits/d4f23ea45e988919def4a546619f9a5d4635d635